### PR TITLE
[KOGITO-515] - Refactoring s2i modules to have a common module

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: Filippe Spolti <fspolti@redhat.com>

### CODE REVIEW

**Review by mswiderski — COMMENTED** [review]
@spolti looks good to me but will leave to @ricardozanini to review in more depth

File: s2i/README.md [hunk 1]
```diff
@@ -3,7 +3,8 @@
 To be able to produce efficient container images for Kogito there are following sets
 of builder images to perform the build and to run the result binaries.
 
-Images are grouped by the runtime that will run the binaries and the OS level
+Images are grouped by the runtime that will run the binaries and the builers which will actually build the
```
Comment (ricardozanini) [inline s2i/README.md]: "builders"

**Review by ricardozanini — CHANGES_REQUESTED** [review]
Great work! Just one minor comment. Aside, I'd like to see some bats tests since this is one of the reasons why we refactored the code. Plus, you told me that we're still missing running this new images on OpenShift. :)
One last thing is that Jekins is complaining. 
Thank you!

File: s2i/README.md (hunk 1 above)
Comment (spolti) [reply s2i/README.md]: nice catch.

**Comment by spolti** [comment]
Sure, will add some bats tests and maybe start to look on run it wit GH Actions. :)

**Review by ricardozanini — APPROVED** [review]

**Comment by ricardozanini** [comment]
Go ahead!

**Comment by ricardozanini** [comment]
@sutaakar could you take a look on the tests please?

File: s2i/modules/kogito-maven/tests/bats/maven-settings.bats [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    echo "expected=${expected}"
+    echo "result=${result}"
+    [ "${expected}" = "${result}" ]
+}
+
+@test "test maven proxy configuration with HTTP_PROXY env custom port http with username" {
+    prepareEnv
+    HTTP_PROXY="http://10.10.10.20:8003"
+    run configure_proxy
+        echo "${lines[@]}"
```
Comment (sutaakar) [inline s2i/modules/kogito-maven/tests/bats/maven-settings.bats]: A leftover from debugging?

Comment (spolti) [reply s2i/modules/kogito-maven/tests/bats/maven-settings.bats]: removed.

File: s2i/modules/kogito-quarkus-jvm/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# JVM Runtime image for Kogito apps
+
+This Runtime image is considered a runtime image as it only cantains the needed
```
Comment (sutaakar) [inline s2i/modules/kogito-quarkus-jvm/README.md]:
```suggestion
This Runtime image is considered a runtime image as it only contains the needed
```

File: s2i/modules/kogito-quarkus-jvm/README.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+files used to execute a pre built Kogito application in native mode.
+
+This is a lightweight and compact application container image.
+
+This image can be used without the image builder, this image have the hability to copy
```
Comment (sutaakar) [inline s2i/modules/kogito-quarkus-jvm/README.md]:
```suggestion
This image can be used without the image builder, this image has the ability to copy
```

File: s2i/modules/kogito-quarkus-jvm/README.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+built artifacts and files into de image so you can perform faster image builds to test your application.
```
Comment (sutaakar) [inline s2i/modules/kogito-quarkus-jvm/README.md]:
```suggestion
built artifacts and files into the image so you can perform faster image builds to test your application.
```

File: s2i/modules/kogito-quarkus/README.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# Native Runtime image for Kogito apps
+
+This Runtime image is considered a runtime image as it only contains the needed
+files used to execute a pre built Kogito application.
+
+This is a lightweight and compact application container image.
+
+This image can be used without the image builder, this image have the hability to copy
```
Comment (sutaakar) [inline s2i/modules/kogito-quarkus/README.md]:
```suggestion
This image can be used without the image builder, this image has the ability to copy
```

File: s2i/modules/kogito-quarkus/README.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+built artifacts and files into de image so you can perform faster image builds to test your application.
```
Comment (sutaakar) [inline s2i/modules/kogito-quarkus/README.md]:
```suggestion
built artifacts and files into the image so you can perform faster image builds to test your application.
```

File: s2i/modules/kogito-springboot/README.md [hunk 8]
```diff
@@ -6,3 +6,11 @@ This S2I is considered a runtime image builder as it relies on another image
 and place into the container.
 
 On top of the OS it has JRE installed (OpenJDK 1.8) to be able to execute fat jar.
+
+This image can be used without the image builder, this image have the hability to copy
```
Comment (sutaakar) [inline s2i/modules/kogito-springboot/README.md]:
```suggestion
This image can be used without the image builder, this image has the ability to copy
```

File: s2i/modules/kogito-springboot/README.md [hunk 9]
```diff
@@ -6,3 +6,11 @@ This S2I is considered a runtime image builder as it relies on another image
...
+built artifacts and files into de image so you can perform faster image builds to test your application.
```
Comment (sutaakar) [inline s2i/modules/kogito-springboot/README.md]:
```suggestion
built artifacts and files into the image so you can perform faster image builds to test your application.
```

**Comment by sutaakar** [comment]
Found few grammar adjustments, otherwise the tests look good.
I haven't tried to build the images locally yet.

**Comment by spolti** [comment]
@sutaakar thanks for the review. PR updated.

**Comment by sutaakar** [comment]
I tried to run the Behave tests locally, but getting:
`     ERROR:root:panic: runtime error: invalid memory address or nil pointer dereference
      ERROR:root:[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x55774e775453]
`
for some tests. Not sure if it is caused by limited memory in my laptop or other issue.
Reproduced using command `cekit -v test --overrides-file kogito-quarkus-s2i-overrides.yaml behave`

**Comment by sutaakar** [comment]
Can somebody else please try to run behave tests?

**Comment by ricardozanini** [comment]
This test should be run by the Jenkins CI. If Jenkins outputs successfully, that means that the behave is ok. Plus, I think @spolti is running those tests before pushing the PR.
Sorry, but I don't have BW to run those tests right now. :-(

**Comment by spolti** [comment]
Behave is passing, the problem I caught was on openshift due a different behavior between manual s2i builds and chained builds on ocp.

**Comment by sutaakar** [comment]
ok, thanks for info.

**Review by ricardozanini — APPROVED** [review]

**Review by sutaakar — APPROVED** [review]
Looks good to me.
Examples are built correctly using rc2 images.

**Comment by spolti** [comment]
Thanks, merging it.

**Pull request merged**